Extensions/TileMap/tilemapruntimeobject.js

```diff
@@ -0,0 +1,242 @@
+  else
+    this._renderer = new gdjs.TileMapRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.TileMapRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.registerObject('TileMap::TileMap', gdjs.TileMapRuntimeObject);
+
+gdjs.TileMapRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+gdjs.TileMapRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._animationSpeed <= 0) return;
+  var elapsedTime = this.getElapsedTime(runtimeScene) / 1000; 
+  
+  this._frameElapsedTime += elapsedTime * this._animationSpeed; 
+  // 0.25 = 4 fps , 0.0625 = 1  fps
+  var normalizedFps = this._animationFps * 0.0625;
+  while ( this._frameElapsedTime > normalizedFps ) {
+    this._renderer.incrementAnimationFrameX();
+    this._frameElapsedTime -= normalizedFps;
+    if ( this._frameElapsedTime < 0 ) this._frameElapsedTime = 0; 
+  }
+  
+};
+/**
+ * @param { TileMapObjectDataType} oldObjectData
+ * @param { TileMapObjectDataType} newObjectData
+ */
+gdjs.TileMapRuntimeObject.prototype.updateFromObjectData = function(
+  oldObjectData,
+  newObjectData
+) {
+  if (oldObjectData.content.opacity !== newObjectData.content.opacity) {
+    this.setOpacity(newObjectData.content.opacity);
+  }
+  if (
+    oldObjectData.content.tilemapJsonFile !==
+    newObjectData.content.tilemapJsonFile
+  ) {
+    this.setTilemapJsonFile(newObjectData.content.tilemapJsonFile);
+  }
+  if (
+    oldObjectData.content.tilemapAtlasImage !==
+    newObjectData.content.tilemapAtlasImage
+  ) {
+    this.setTilemapAtlasImage(newObjectData.content.tilemapAtlasImage);
+  }
+  if (oldObjectData.content.displayMode !== newObjectData.content.displayMode) {
+    this.setDisplayMode(newObjectData.content.displayMode);
+  }
+  if (oldObjectData.content.layerIndex !== newObjectData.content.layerIndex) {
+    this.setLayerIndex(newObjectData.content.layerIndex);
+  }
+  if (
+    oldObjectData.content.animationSpeed !==
+    newObjectData.content.animationSpeed
+  ) {
+    this.setAnimationSpeed(newObjectData.content.animationSpeed);
+  }
+  if (
+    oldObjectData.content.animationFps !== newObjectData.content.animationFps
+  ) {
+    this.setAnimationFps(newObjectData.content.animationFps);
+  }
+
+  return true;
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.TileMapRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setHeight(initialInstanceData.height);
+  }
+};
+
+gdjs.TileMapRuntimeObject.prototype.onDestroyFromScene = function(
+  runtimeScene
+) {
+  gdjs.RuntimeObject.prototype.onDestroyFromScene.call(this, runtimeScene);
+};
+
+/**
+ * Set/Get TileMap base properties
```

Comment:
A JSDoc comment for a function (/** ... */) should only be about the function.
Here: `Set the Tilemap json file to display.`